### PR TITLE
Optimize struct handling in MulitplicativeSplitDefgradElastHyper material

### PR DESCRIPTION
**Pull request by dragos-ana**
## Description and Context

The current management of the struct instance `kinematic_quantities` in `src/mat/4C_mat_multiplicative_split_defgrad_elasthyper.cpp` is not optimal, as some of its variables are modified during the associated method calls, while others should remain `const` throughout.  Additionally, this struct also contains both elastic stiffness components and stress factors, which are not kinematic quantities. This PR addresses these issues, and  the following work has been performed:

- Removed the elastic stiffness components from `KinematicQuantities`.
- Removed the $\gamma_i$ and $\delta_i$ stress factors  from `KinematicQuantities` and added them to a separate `StressFactors` struct. 
- Aside from `evaluate_kin_quant_elast`, which evaluates all of its variables, the struct instance of `KinematicQuantities` is now only passed by const reference to the material methods. The same applies for the struct instance of `StressFactors`. Furthermore,  `evaluated_s_d_ifin` and `evaluate_additional_cmat` have been reformulated to return the computed objects directly.

### CODE REVIEW

File: unittests/mat/4C_multiplicative_split_defgrad_elasthyper_test.cpp [hunk 1]
```diff
@@ -417,8 +417,7 @@ namespace
     const double concentration(44327.362);
     set_concentration_to_inelastic_material(concentration);
 
-    // actual call that is tested
-    Core::LinAlg::Matrix<6, 6> cMatAdd(true);
+    ;
```
Comment (sebproell) [inline unittests/mat/4C_multiplicative_split_defgrad_elasthyper_test.cpp:420]:
```suggestion
```

**Review by sebproell — COMMENTED** [review]
Can you squash the meaningless second commit?

**Comment by dragos-ana** [comment]
> Can you squash the meaningless second commit?

Done. Thanks for the review, @sebproell !

**Review by sebproell — APPROVED** [review]

**Review by c-p-schmidt — APPROVED** [review]
Looks good to me.

**Pull request merged**